Content.Server/GameTicking/Rules/ZombieRuleSystem.cs

```diff
@@ -0,0 +1,308 @@
+        if (!ev.Forced && ev.Players.Length < minPlayers)
+        {
+            _chatManager.DispatchServerAnnouncement(Loc.GetString("zombie-not-enough-ready-players", ("readyPlayersCount", ev.Players.Length), ("minimumPlayers", minPlayers)));
+            ev.Cancel();
+            return;
+        }
+
+        if (ev.Players.Length == 0)
+        {
+            _chatManager.DispatchServerAnnouncement(Loc.GetString("zombie-no-one-ready"));
+            ev.Cancel();
+            return;
+        }
+    }
+
+    public override void Started(GameRuleConfiguration configuration)
+    {
+        //this technically will run twice with zombies on roundstart, but it doesn't matter because it fails instantly
+        InfectInitialPlayers();
+    }
+
+    public override void Ended(GameRuleConfiguration configuration) { }
+
+    private void OnZombifySelf(EntityUid uid, ZombifyOnDeathComponent component, ZombifySelfActionEvent args)
+    {
+        _zombify.ZombifyEntity(uid);
+
+        var action = new InstantAction(_prototypeManager.Index<InstantActionPrototype>(ZombifySelfActionPrototype));
+        _action.RemoveAction(uid, action);
+    }
+
+    private float GetInfectedPercentage(out List<EntityUid> livingHumans)
+    {
+        var allPlayers = EntityQuery<HumanoidAppearanceComponent, MobStateComponent>();
+        var totalPlayers = new List<EntityUid>();
+        var livingZombies = new List<EntityUid>();
+        livingHumans = new();
+        foreach (var ent in allPlayers)
+        {
+            if (ent.Item2.IsAlive())
+            {
+                totalPlayers.Add(ent.Item2.Owner);
+
+                if (HasComp<ZombieComponent>(ent.Item1.Owner))
+                    livingZombies.Add(ent.Item2.Owner);
+                else
+                    livingHumans.Add(ent.Item2.Owner);
+            }
+        }
+        return livingZombies.Count / totalPlayers.Count;
+    }
+
+    /// <summary>
+    ///     Infects the first players with the passive zombie virus.
+    ///     Also records their names for the end of round screen.
+    /// </summary>
+    /// <remarks>
+    ///     The reason this code is written separately is to facilitate
+    ///     allowing this gamemode to be started midround. As such, it doesn't need
+    ///     any information besides just running.
+    /// </remarks>
+    private void InfectInitialPlayers()
+    {
+        var allPlayers = _playerManager.ServerSessions.ToList();
+        var playerList = new List<IPlayerSession>();
+        var prefList = new List<IPlayerSession>();
+        foreach (var player in allPlayers)
+        {
+            if (player.AttachedEntity != null)
+            {
+                playerList.Add(player);
+
+                var pref = (HumanoidCharacterProfile) _prefs.GetPreferences(player.UserId).SelectedCharacter;
+                if (pref.AntagPreferences.Contains(PatientZeroPrototypeID))
+                    prefList.Add(player);
+            }
+        }
```

Comment:
Would this try to infect people in lobby?

Reply:
No, because the AttachedEntity != check fails, removing them from the playerList pool. I tested this multiple times locally and couldn't get it to happen once.